lib/ansible/modules/network/nxos/nxos_vsan.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+
+DOCUMENTATION = '''
+module: nxos_vsan
+extends_documentation_fragment: nxos
+version_added: 2.9
+short_description: Configuration of vsan.
+description:
+    - Configuration of vsan for Cisco MDS NXOS.
+author:
+    - Suhas Bharadwaj (@srbharadwaj) (subharad@cisco.com)
+options:
+    vsan:
+        description:
+            - List of vsan details to be added or removed
+        type: list
+        suboptions:
+            id:
+                description:
+                    - Vsan id
+                required: True
+            name:
+                description:
+                    - Name of the vsan
+            suspend:
+                description:
+                    - suspend the vsan if True
+                type: bool
+                default: False
+            remove:
+                description:
+                    - Removes the vsan if True
+                type: bool
+                default: False
+            interface:
+                description:
+                    - List of vsan's interfaces to be added
+                type: list
+'''
+
+EXAMPLES = '''
+---
+-
+  name: "Test that vsan module works"
+  nxos_vsan:
+    provider: "{{ creds }}"
```

Comment:
Please remove `provider` as we don't plan to support connection local for newly added module.